pkg/util/lifted/scheduler/cache/node_tree_test.go

```diff
@@ -0,0 +1,154 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// This code is directly lifted from the Kubernetes codebase in order to avoid relying on the k8s.io/kubernetes package.
+// For reference:
+// https://github.com/kubernetes/kubernetes/blob/release-1.26/pkg/scheduler/internal/cache/node_tree.go
```

Comment:
Hi @yashpandey06, I assume you copied it from this file: https://github.com/kubernetes/kubernetes/blob/release-1.26/pkg/scheduler/internal/cache/node_tree_test.go ?

Reply:
Hey @XiShanYongYe-Chang  , i took it from node_tree itself for which the tests have been written . please see

<img width="971" alt="Screenshot 2024-08-26 at 9 15 13 AM" src="https://github.com/user-attachments/assets/17cb4fa7-c33d-4b32-bb69-fa8a535f9582">

Reply:
Aren't we suppose to take headers from the file we are writing tests for ?

Reply:
yup you can say

Reply:
> Hey @XiShanYongYe-Chang , i took headers from node_tree itself for which the tests have been written . please see
> 
> <img alt="Screenshot 2024-08-26 at 9 15 13 AM" width="971" src="https://private-user-images.githubusercontent.com/97700473/361298982-17cb4fa7-c33d-4b32-bb69-fa8a535f9582.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ2NDQyNjYsIm5iZiI6MTcyNDY0Mzk2NiwicGF0aCI6Ii85NzcwMDQ3My8zNjEyOTg5ODItMTdjYjRmYTctYzMzZC00YjMyLWJiNjktZmE4YTUzNWY5NTgyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODI2VDAzNDYwNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY0ODhjMjE3MDc5YjJmZTYyZDNmZDI2MTYxYTRiNTViOTQ4NGRmZTQ4MDY4YmRkNjFhOTJiNWNiZWNiNmY2MWUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.enJaynDzosqysfWK3XoYDAfsQu6F0QeMS0brqyn_Hlo">

Reply:
Let me know  if it clears the doubt , I ll resolve this thread !

Reply:
I mean you can just copy the test functions you need from [this file](https://github.com/kubernetes/kubernetes/blob/release-1.26/pkg/scheduler/internal/cache/node_tree_test.go), no need to add extra tests yourself.

Reply:
@XiShanYongYe-Chang I got  your point .Let me revisit this .